test/e2e/upgrade/upgrade_test.go

```diff
@@ -0,0 +1,126 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/iancoleman/strcase"
+	ispnv1 "github.com/infinispan/infinispan-operator/api/v1"
+	"github.com/infinispan/infinispan-operator/api/v2alpha1"
+	"github.com/infinispan/infinispan-operator/controllers"
+	"github.com/infinispan/infinispan-operator/controllers/constants"
+	batchtest "github.com/infinispan/infinispan-operator/test/e2e/batch"
+	tutils "github.com/infinispan/infinispan-operator/test/e2e/utils"
+	coreos "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	testKube = tutils.NewTestKubernetes(os.Getenv("TESTING_CONTEXT"))
+	// startChannel          = os.Getenv("SUBSCRIPTION_CHANNEL_START")
+	subName               = constants.GetEnvWithDefault("SUBSCRIPTION_NAME", "infinispan-operator")
+	subNamespace          = constants.GetEnvWithDefault("SUBSCRIPTION_NAMESPACE", tutils.Namespace)
+	sourceChannel         = constants.GetEnvWithDefault("SUBSCRIPTION_CHANNEL_SOURCE", "2.1.x")
+	latestChannel         = constants.GetEnvWithDefault("SUBSCRIPTION_CHANNEL_LATEST", "2.2.x")
```

Comment:
This value can be parsed so it doesn't need to be updated with each new release

Reply:
Originally I was thinking of just having this as a ENV var without a default and then setting this centrally for CI. However, that makes it more painful to run locally.

Any thoughts on where we could parse the value from for it to work with the centralised Makefile and go testsuite?

Reply:
https://github.com/infinispan/infinispan-operator/pull/1211/files#diff-b2218cd9f8b91e8747cc3b42a41c88f4e770132ba2a117f616c44b61fceac3d4R12-R49 is piece of code in java that covers what you are looking for. It shouldn't be an issue to get the same in go on test init.

Reply:
I have updated it to automatically populate the default value, but it's also possible to override the source/target channels via the `SUBSCRIPTION_CHANNEL_SOURCE` and `SUBSCRIPTION_CHANNEL_TARGET` env variables respectively.